src/main/java/com/epam/reportportal/auth/config/SamlServiceProviderBeanConfiguration.java

```diff
@@ -0,0 +1,118 @@
+import com.epam.reportportal.auth.integration.saml.SamlAuthenticationSuccessHandler;
+import com.epam.reportportal.auth.integration.saml.SamlUserReplicator;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.saml.SamlKeyException;
+import org.springframework.security.saml.key.SimpleKey;
+import org.springframework.security.saml.provider.SamlServerConfiguration;
+import org.springframework.security.saml.provider.service.authentication.GenericErrorAuthenticationFailureHandler;
+import org.springframework.security.saml.provider.service.authentication.SamlAuthenticationResponseFilter;
+import org.springframework.security.saml.provider.service.config.SamlServiceProviderServerBeanConfiguration;
+import org.springframework.security.saml.spi.SamlKeyStoreProvider;
+import org.springframework.security.saml.spi.SpringSecuritySaml;
+import org.springframework.security.saml.spi.opensaml.OpenSamlImplementation;
+import org.springframework.security.saml.util.X509Utilities;
+
+import javax.servlet.Filter;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Base64;
+
+import static org.springframework.util.StringUtils.hasText;
+
+/**
+ * Bean declarations for service provider part of SAML integration
+ *
+ * @author Yevgeniy Svalukhin
+ */
+@Configuration
+public class SamlServiceProviderBeanConfiguration extends SamlServiceProviderServerBeanConfiguration {
+
+    private SamlAuthenticationSuccessHandler samlSuccessHandler;
+    private SamlUserReplicator samlUserReplicator;
+    private SamlServerConfiguration serviceProviderConfiguration;
+
+    public SamlServiceProviderBeanConfiguration(SamlServerConfiguration spConfiguration, SamlAuthenticationSuccessHandler samlSuccessHandler,
+                                                SamlUserReplicator samlUserReplicator) {
+        this.serviceProviderConfiguration = spConfiguration;
+        this.samlSuccessHandler = samlSuccessHandler;
+        this.samlUserReplicator = samlUserReplicator;
+    }
+
+    @Override
+    protected SamlServerConfiguration getDefaultHostSamlServerConfiguration() {
+        return serviceProviderConfiguration;
+    }
+
+    @Override
+    public Filter spAuthenticationResponseFilter() {
+        SamlAuthenticationResponseFilter authenticationFilter = new SamlAuthenticationResponseFilter(getSamlProvisioning());
+        authenticationFilter.setAuthenticationManager(new ReportPortalSamlAuthenticationManager(samlUserReplicator));
+        authenticationFilter.setAuthenticationSuccessHandler(samlSuccessHandler);
+        authenticationFilter.setAuthenticationFailureHandler(new GenericErrorAuthenticationFailureHandler());
```

Comment:
i'm not really sure that will work with our UI. 
As i remember, in case of error, we redirect back to UI with error message as query parameter.

Reply:
Failure handler has been extracted to dedicated class from OAuthSecurityConfig and it has been used in configuration of filter for SAML